table/iterator.go

```diff
@@ -83,6 +88,16 @@ func (itr *blockIterator) setIdx(i int) {
 		// EndOffset of the current entry is the start offset of the next entry.
 		endOffset = int(itr.entryOffsets[itr.idx+1])
 	}
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Printf("==== Recovered====\n%s", string(debug.Stack()))
+			fmt.Printf("Table ID: %d\nBlock ID: %d\nEntry Idx: %d\nData len: %d\n"+
+				"StartOffset: %d\nEndOffset: %d\nEntryOffsets len: %d\nEntryOffsets: %v\n",
+				itr.tableID, itr.blockID, itr.idx, len(itr.data), startOffset, endOffset,
+				len(itr.entryOffsets), itr.entryOffsets)
+			os.Exit(1)
```

Comment:
Do you need `os.Exit()` here? Won't it prevent the event from being captured to sentry?

Reply:
Yes. I can panic instead. Thanks